_posts/2020-11-13-python-forking-vs-spawn.md

```diff
@@ -169,9 +169,14 @@ import datetime as dt
 import logging
 from multiprocessing import get_context
 import os
+import sys
 import time
 from threading import Lock, Thread, enumerate
 
+if sys.platform in ('win32', 'msys', 'cygwin'):
+    print("Script works only on *NIX type operating systems.")
+    sys.exit(1)
+
```

Comment:
I suppose if we wanted some windows compatibility, we could have a context based on the platform eg. ['spawn'] for windows and ['spawn', 'fork'] for nix. (Although maybe that's making it too complicated and missing the point of the post.)

Reply:
I had weirdness when I ran it in Spyder in Windows, too.  🤷

The main aim is to show the difference between fork and spawn and explain why the hanging thread issue happens.  Windows doesn't have fork or the hanging thread problem, so I think it's OK just to stop the script at the start for them.

Reply:
yes, that seems reasonable 👍